core/src/main/java/com/homeaway/streamplatform/streamregistry/UUIDMDCRequestFilter.java

```diff
@@ -0,0 +1,33 @@
+/* Copyright (c) 2018 Expedia Group.
+ * All rights reserved.  http://www.homeaway.com
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *      http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.homeaway.streamplatform.streamregistry;
+
+import java.util.UUID;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
```

Comment:
added the request UUID for each http request, so that grouping and debugging logs are easy.

Sample log for inserting a Stream with UUID in each log statement,

```
[11:12:46.971] [dw-26 - PUT /v0/streams/TestMDCStream] INFO   c.h.s.s.e.s.ConfluentSchemaManager b69e9f65-1de5-4f54-82a3-f4c1410694e5 - Schema registration successful. Subject=TestMDCStream-key ; id=3 ; version=1
[11:12:46.977] [dw-26 - PUT /v0/streams/TestMDCStream] INFO   c.h.s.s.e.s.ConfluentSchemaManager b69e9f65-1de5-4f54-82a3-f4c1410694e5 - Schema registration successful. Subject=TestMDCStream-value ; id=3 ; version=1
[11:12:47.065] [dw-26 - PUT /v0/streams/TestMDCStream] INFO   c.h.s.s.d.d.i.StreamDaoImpl b69e9f65-1de5-4f54-82a3-f4c1410694e5 - key={"streamName": "TestMDCStream"} available for the stream-name=TestMDCStream
[11:12:47.065] [dw-26 - PUT /v0/streams/TestMDCStream] INFO   c.h.s.s.d.d.i.StreamDaoImpl b69e9f65-1de5-4f54-82a3-f4c1410694e5 - creating topics for vpcList: [vpc-localhost]
[11:12:47.065] [dw-26 - PUT /v0/streams/TestMDCStream] INFO   c.h.s.s.d.d.AbstractDao b69e9f65-1de5-4f54-82a3-f4c1410694e5 - Cluster Details: vpc: vpc-localhost, env: test, hint: primary, actorType: producer
[11:12:47.066] [dw-26 - PUT /v0/streams/TestMDCStream] DEBUG  c.h.s.s.p.i.KafkaInfraManager b69e9f65-1de5-4f54-82a3-f4c1410694e5 - Approximate Num. of Entries in Infra Table-2
[11:12:47.077] [dw-26 - PUT /v0/streams/TestMDCStream] INFO   c.h.s.s.p.i.KafkaInfraManager b69e9f65-1de5-4f54-82a3-f4c1410694e5 - Cluster Name - {cluster.name=cluster-localhost-primary, schema.registry.url=http://localhost:8081, bootstrap.servers=localhost:9092, zookeeper.quorum=localhost:2181}
[11:12:47.077] [dw-26 - PUT /v0/streams/TestMDCStream] INFO   c.h.s.s.d.d.AbstractDao b69e9f65-1de5-4f54-82a3-f4c1410694e5 - Cluster Information found - Optional[{"clusterProperties": {"cluster.name": "cluster-localhost-primary", "schema.registry.url": "http://localhost:8081", "bootstrap.servers": "localhost:9092", "zookeeper.quorum": "localhost:2181"}}]
[11:12:47.264] [dw-26 - PUT /v0/streams/TestMDCStream] DEBUG  c.h.s.s.d.d.i.KafkaManagerImpl b69e9f65-1de5-4f54-82a3-f4c1410694e5 - topic: TestMDCStream exists=true
[11:12:47.358] [dw-26 - PUT /v0/streams/TestMDCStream] INFO   c.h.s.s.d.d.i.KafkaManagerImpl b69e9f65-1de5-4f54-82a3-f4c1410694e5 - topic config for TestMDCStream exactly match. Ignoring.
[11:12:47.358] [ZkClient-EventThread-37-localhost:2181] INFO   o.I.z.ZkEventThread  - Terminate ZkClient event thread.
[11:12:47.361] [dw-26 - PUT /v0/streams/TestMDCStream] INFO   o.a.z.ZooKeeper b69e9f65-1de5-4f54-82a3-f4c1410694e5 - Session: 0x1000154b8d9002a closed
[11:12:47.361] [dw-26 - PUT /v0/streams/TestMDCStream] INFO   c.h.s.s.d.d.i.StreamDaoImpl b69e9f65-1de5-4f54-82a3-f4c1410694e5 - Topic TestMDCStream created/updated at localhost:9092
[11:12:47.361] [dw-26 - PUT /v0/streams/TestMDCStream-EventThread] INFO   o.a.z.ClientCnxn  - EventThread shut down for session: 0x1000154b8d9002a
[11:12:47.409] [dw-26 - PUT /v0/streams/TestMDCStream] INFO   c.h.s.s.s.ManagedKafkaProducer b69e9f65-1de5-4f54-82a3-f4c1410694e5 - Message pushed to the sourceKStreamProcessorTopic Topic=_streamregistry_streams_v0.1 with key={"streamName": "TestMDCStream"} successfully
[11:12:47.409] [dw-26 - PUT /v0/streams/TestMDCStream] INFO   c.h.s.s.d.d.i.StreamDaoImpl b69e9f65-1de5-4f54-82a3-f4c1410694e5 - Stream upserted for TestMDCStream```